tests/unit/core/test_searchUtils.js

```diff
@@ -171,4 +177,70 @@ describe(__filename, () => {
       expect(osFilterValue).toEqual(undefined);
     });
   });
+
+  describe('fixFiltersForAndroidThemes', () => {
+    it('changes clientApp filter to `firefox` for Android themes', () => {
+      const filters = {
+        addonType: ADDON_TYPE_THEME,
+        clientApp: CLIENT_APP_ANDROID,
+      };
+
+      const newFilters = fixFiltersForAndroidThemes({ filters });
+      expect(newFilters).toEqual({
+        ...filters,
+        clientApp: CLIENT_APP_FIREFOX,
+      });
+    });
+
+    it('does not change clientApp filter for Android extensions', () => {
+      const filters = {
+        addonType: ADDON_TYPE_EXTENSION,
+        clientApp: CLIENT_APP_ANDROID,
+      };
+
+      const newFilters = fixFiltersForAndroidThemes({ filters });
+      expect(newFilters).toEqual(filters);
+    });
+
+    it('does not do anything when clientApp is `firefox`', () => {
+      const filters = {
+        addonType: ADDON_TYPE_THEME,
+        clientApp: CLIENT_APP_FIREFOX,
```

Comment:
This isn't a great test because it will never fail. You could make it fail by changing the filter to `clientApp: 'some-bogus-client'` and writing the spec more like "it does not change filters unless clientApp is `android`"

Reply:
ah indeed, thanks!